canonical.go

```diff
@@ -117,12 +117,8 @@ func (c *relaxedBodyCanonicalizer) Write(b []byte) (int, error) {
 			if len(c.crlfBuf) > 0 {
 				canonical = append(canonical, c.crlfBuf...)
 				c.crlfBuf = nil
-
-				if len(c.wspBuf) > 0 {
-					canonical = append(canonical, c.wspBuf...)
-					c.wspBuf = nil
-				}
```

Comment:
Why is this part removed? We don't always want to add only a single space.

Reply:
https://tools.ietf.org/html/rfc6376#section-3.4 
3.4.4.  The "relaxed" Body Canonicalization Algorithm
 Reduce all sequences of WSP within a line to a single SP
          character.

Reply:
Okay, I was confused by the term "within a line", assuming at the beginning of a line was not within a line.

Please fix the test then!